libathemecore/ptasks.c

```diff
@@ -567,17 +567,34 @@ handle_message(struct sourceinfo *si, char *target, bool is_notice, char *messag
 	if (si->su == NULL)
 		return;
 
-	/* if this is a channel, handle it differently. */
-	if (*target == '#')
+	// If the target is a channel, handle it differently
+	if ((p = strchr(target, '#')) != NULL)
 	{
-		handle_channel_message(si, target, is_notice, message);
-		return;
+		/* Account for the fact that the message may have a status prefix on it; for example, a message
+		 * to @#channel should still be treated as a message to #channel rather than ignored.   -- amdj
+		 */
+		bool is_status_prefixed = false;
+
+		for (unsigned int i = 0; p == (target + 1) && prefix_mode_list && prefix_mode_list[i].mode; i++)
```

Comment:
i feel like the inclusion of static logic in a for condition is needlessly abstruce. this took me 5 minutes to understand

Reply:
is there much at risk to just do

```c
if ((p = strchr(target, '#')) != NULL)
{
    (void) handle_channel_message(si, p, is_notice, message);
    return;
}
```

?

it feels like having an octothorpe in a target name when it's not ultimately a channel is a protocol framing issue

Reply:
This was proposed about a decade ago, but there was something wrong with it.  I forget what it was now, though.

Honestly, it seems fine, we were probably bikeshedding about a non-practical issue.

Reply:
> is there much at risk to just do
> 
> ```c
> if ((p = strchr(target, '#')) != NULL)
> {
>     (void) handle_channel_message(si, p, is_notice, message);
>     return;
> }
> ```

Think of charybdis-family IRCds where you send a PRIVMSG to `$#some.mask` and `#some.mask` happens to be an extant channel.